contracts/src/main.sw

```diff
@@ -278,62 +238,62 @@ impl Exchange for Contract {
         (eth_amount, token_amount)
     }
 
-    fn swap_with_minimum(gas_: u64, amount_: u64, asset_id_: b256, params: SwapWithMinimumParams) -> u64 {
-        assert(params.deadline >= height());
-        assert(msg_amount() > 0 && params.min > 0);
-        assert(msg_asset_id() == ETH_ID || msg_asset_id() == TOKEN_ID);
+    fn swap_with_minimum(min: u64, deadline: u64) -> u64 {
+        assert(deadline >= height());
+        assert(msg_amount() > 0 && min > 0);
+        assert((msg_asset_id()).into() == ETH_ID || (msg_asset_id()).into() == TOKEN_ID);
 
         // TODO replace
         // let sender = msg_sender().unwrap();
         let sender = ~Address::from(0x0000000000000000000000000000000000000000000000000000000000000000);
 
-        let eth_reserve = this_balance(ETH_ID);
-        let token_reserve = this_balance(TOKEN_ID);
+        let eth_reserve = this_balance(~ContractId::from(ETH_ID));
```

Comment:
Can you file an issue to make these constants `ContractId`s (unless there's already one)?

Reply:
You mean something like `const ETH_ID = ~ContractId::from(0x00..00)`? Yeah I think this requires compiler support as we only expect literals on the RHS currently. Let me file an issue.

Reply:
Yes